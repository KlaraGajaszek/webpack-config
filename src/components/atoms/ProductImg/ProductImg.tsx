import React, { FC } from 'react';

import { PropsProductImage } from '../../../models/Product';

const ProductImg: FC<PropsProductImage> = ({ imageSrc }) => {
  return (
    <img
      className="destination-card-image"
      data-testid="imageID"
      src={imageSrc}
      alt="description of image"
    />
  );
};

export default ProductImg;
