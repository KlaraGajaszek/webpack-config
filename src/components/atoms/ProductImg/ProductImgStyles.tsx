import styled from 'styled-components';

export const ProductImage = styled.img`
  height: 100%;
  width: 100%;
  border-radius: 8px;
`;
