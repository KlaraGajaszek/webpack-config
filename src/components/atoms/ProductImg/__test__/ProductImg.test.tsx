import React from 'react'
import 'jest-styled-components'

import { screen, render, within } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import renderer from 'react-test-renderer'

import ProductImg from '../ProductImg'

describe('ProductImg', () => {
    it('render without crash', () => {
        const src = 'https://picsum.photos/1000/300'
        render(<ProductImg imageSrc={src} />)
    })
    it('check if src render correct', () => {
        const src = 'https://picsum.photos/1000/300'
        const imageProduct = renderer
            .create(<ProductImg imageSrc={src} />)
            .toJSON()
        expect(imageProduct).toMatchSnapshot()
    })
})
