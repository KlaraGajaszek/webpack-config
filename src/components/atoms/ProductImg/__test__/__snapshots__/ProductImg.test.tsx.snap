// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImg check if src render correct 1`] = `
<img
  alt="description of image"
  className="destination-card-image"
  data-testid="imageID"
  src="https://picsum.photos/1000/300"
/>
`;
